Fix plan for google_identity_platform_config always showing a change for phone_number.enabled=false and email.enabled=false

pashanka opened the pull request:
Fixes issue https://github.com/hashicorp/terraform-provider-google/issues/15043

**Release Note Template for Downstream PRs (will be copied)**

```release-note: bug
identityplatform: Fixed perma-diff in `google_identity_platform_config`
```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12255-old..auto-pr-12255) ( 1 file changed, 8 insertions(+), 17 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12255-old..auto-pr-12255) ( 1 file changed, 8 insertions(+), 17 deletions(-))

rainshen49 commented on mmv1/templates/terraform/custom_flatten/identity_platform_config_phone_number.go.tmpl:
1. I notice this is different from https://github.com/rainshen49/magic-modules/blame/main/mmv1/templates/terraform/custom_flatten/identity_platform_config_anonymous.go.tmpl#L13, in particular, if the entire value is nil. Is there a motivation for why?
2. Pls figure out indentation situation. Most likely there is a mix of tabs and spaces.

pashanka replied:
1. Yes this is slightly different because, I observed that in case of anonymous field the response comes as anonymous: {}, where as in case of phone_number/email v itself seems to be nil.
2. Fixed indentation.

rainshen49 commented:
Do you want to improve the code quality by adding a test? Currently, none of the tests cover this scenario, which is why the bug occurred.

```terraform
anonymous {
        enabled = false
    }
    email {
        enabled = false
    }
    phone_number {
        enabled = false
    }
```

I can walk you through how to create a new test, but it's basically making a new `example` https://github.com/rainshen49/magic-modules/blob/main/mmv1/products/identityplatform/Config.yaml#L47

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12255-old..auto-pr-12255) ( 2 files changed, 71 insertions(+), 17 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12255-old..auto-pr-12255) ( 2 files changed, 71 insertions(+), 17 deletions(-))

modular-magician commented:
#### Tests analytics
Total tests: 17
Passed tests: 14
Skipped tests: 2
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>identityplatform</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
🔴 Tests failed during RECORDING mode:
`TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/b36cf6cd-d3f3-4e61-bebc-da59b3e67aeb/build-log/recording_build/TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/b36cf6cd-d3f3-4e61-bebc-da59b3e67aeb/recording/TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/b36cf6cd-d3f3-4e61-bebc-da59b3e67aeb/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/b36cf6cd-d3f3-4e61-bebc-da59b3e67aeb/recording) for each test

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12255-old..auto-pr-12255) ( 2 files changed, 71 insertions(+), 17 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12255-old..auto-pr-12255) ( 2 files changed, 71 insertions(+), 17 deletions(-))

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12255-old..auto-pr-12255) ( 2 files changed, 71 insertions(+), 17 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12255-old..auto-pr-12255) ( 2 files changed, 71 insertions(+), 17 deletions(-))

modular-magician commented:
#### Tests analytics
Total tests: 17
Passed tests: 14
Skipped tests: 2
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>identityplatform</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
#### Tests analytics
Total tests: 17
Passed tests: 14
Skipped tests: 2
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>identityplatform</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
🔴 Tests failed during RECORDING mode:
`TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/1213286b-3b32-4465-a919-e3045e93943b/build-log/recording_build/TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/1213286b-3b32-4465-a919-e3045e93943b/recording/TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/1213286b-3b32-4465-a919-e3045e93943b/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/1213286b-3b32-4465-a919-e3045e93943b/recording) for each test

modular-magician commented:
🔴 Tests failed during RECORDING mode:
`TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample` [[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/436e8241-d49b-449d-ab6d-dbdf794e1bda/build-log/recording_build/TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/436e8241-d49b-449d-ab6d-dbdf794e1bda/recording/TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample.log)]



🔴 Errors occurred during RECORDING mode. Please fix them to complete your PR.



View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/436e8241-d49b-449d-ab6d-dbdf794e1bda/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/436e8241-d49b-449d-ab6d-dbdf794e1bda/recording) for each test

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12255-old..auto-pr-12255) ( 2 files changed, 95 insertions(+), 34 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12255-old..auto-pr-12255) ( 2 files changed, 95 insertions(+), 34 deletions(-))

modular-magician commented:
#### Tests analytics
Total tests: 17
Passed tests: 14
Skipped tests: 2
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>identityplatform</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/9f3c9810-c197-48d6-a8f8-800f46ae7f7f/recording/TestAccIdentityPlatformConfig_identityPlatformConfigWithFalseValuesExample.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---






🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/9f3c9810-c197-48d6-a8f8-800f46ae7f7f/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-12255/artifacts/9f3c9810-c197-48d6-a8f8-800f46ae7f7f/recording) for each test

pashanka commented:
> Do you want to improve the code quality by adding a test? Currently, none of the tests cover this scenario, which is why the bug occurred.
> 
> ```hcl
> anonymous {
>     enabled = false
> }
> email {
>     enabled = false
> }
> phone_number {
>     enabled = false
> }
> ```
> 
> I can walk you through how to create a new test, but it's basically making a new `example` https://github.com/rainshen49/magic-modules/blob/main/mmv1/products/identityplatform/Config.yaml#L47

Good point, thanks. Added test and identified email also has similar issue. Fixed that as well.
